### PR TITLE
Update deployment runners

### LINKED ISSUE #1870: [Feature] -d flag to autonomy run so that shell doesnt have to left unattended or managed with tmux
**Is your feature request related to a problem? Please describe.**
At present, autonomy run is persistent, ie. when you run you run docker in the foreground. 

**Describe the solution you'd like**
If we have a `-d` flag  to the run command, then we can run the service and continue to run our shell
This just needs to be added to the docker run command

### LINKED ISSUE #1911: Clean exit from `autonomy deploy run` failure
**Is your feature request related to a problem? Please describe.**
We currently serve an unnecessary stack trace:
```
Running build @ /Users/davidminarsch/v_projects/governatooorr/governatooorr/abci_build
Recreating abci0 ... done

ERROR: for node0  Container "248124df7ff0" is unhealthy.
Traceback (most recent call last):
  File "/Users/davidminarsch/.local/share/virtualenvs/governatooorr-P6Krhoo6/bin/autonomy", line 8, in <module>
    sys.exit(cli())
  File "/Users/davidminarsch/.local/share/virtualenvs/governatooorr-P6Krhoo6/lib/python3.10/site-packages/click/core.py", line 1126, in __call__
    return self.main(*args, **kwargs)
  File "/Users/davidminarsch/.local/share/virtualenvs/governatooorr-P6Krhoo6/lib/python3.10/site-packages/click/core.py", line 1051, in main
    rv = self.invoke(ctx)
  File "/Users/davidminarsch/.local/share/virtualenvs/governatooorr-P6Krhoo6/lib/python3.10/site-packages/click/core.py", line 1657, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/davidminarsch/.local/share/virtualenvs/governatooorr-P6Krhoo6/lib/python3.10/site-packages/click/core.py", line 1657, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/davidminarsch/.local/share/virtualenvs/governatooorr-P6Krhoo6/lib/python3.10/site-packages/click/core.py", line 1393, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/davidminarsch/.local/share/virtualenvs/governatooorr-P6Krhoo6/lib/python3.10/site-packages/click/core.py", line 752, in invoke
    return __callback(*args, **kwargs)
  File "/Users/davidminarsch/.local/share/virtualenvs/governatooorr-P6Krhoo6/lib/python3.10/site-packages/autonomy/cli/deploy.py", line 246, in run
    run_deployment(build_dir, no_recreate, remove_orphans)
  File "/Users/davidminarsch/.local/share/virtualenvs/governatooorr-P6Krhoo6/lib/python3.10/site-packages/autonomy/cli/helpers/deployment.py", line 96, in run_deployment
    commands.up(
  File "/Users/davidminarsch/.local/share/virtualenvs/governatooorr-P6Krhoo6/lib/python3.10/site-packages/compose/metrics/decorator.py", line 18, in wrapper
    result = fn(*args, **kwargs)
  File "/Users/davidminarsch/.local/share/virtualenvs/governatooorr-P6Krhoo6/lib/python3.10/site-packages/compose/cli/main.py", line 1186, in up
    to_attach = up(False)
  File "/Users/davidminarsch/.local/share/virtualenvs/governatooorr-P6Krhoo6/lib/python3.10/site-packages/compose/cli/main.py", line 1166, in up
    return self.project.up(
  File "/Users/davidminarsch/.local/share/virtualenvs/governatooorr-P6Krhoo6/lib/python3.10/site-packages/compose/project.py", line 705, in up
    raise ProjectError(
compose.project.ProjectError: Encountered errors while bringing up the project.
```

**Describe the solution you'd like**
Catch error and display error message cleanly

### LINKED ISSUE #2031: Make sure the containers are stopped properly when exiting from deployment

### PR DESCRIPTION
**Pull request by angrybayblade**
## Proposed changes

This PR 

- Adds support for running services in background
- Implements clean exit from a service
- Adds support for stopping services running in background

## Fixes

Fixes #2031 
Fixes #1870 
Fixes #1911 

## Types of changes

What types of changes does your code introduce? (A **breaking change** is a fix or feature that would cause existing functionality and APIs to not work as expected.)
_Put an `x` in the box that applies_

- [ ] Non-breaking fix (non-breaking change which fixes an issue)
- [ ] Breaking fix (breaking change which fixes an issue)
- [ ] Non-breaking feature (non-breaking change which adds functionality)
- [ ] Breaking feature (breaking change which adds functionality)
- [ ] Refactor (non-breaking change which changes implementation)
- [ ] Messy (mixture of the above - requires explanation!)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../blob/main/CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `main` branch (left side). Also you should start your branch off our `main`.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have locally run services that could be impacted and they do not present failures derived from my changes
- [ ] Public-facing documentation has been updated with the changes affected by this PR. Even if the provided contents are not in their final form, all significant information must be included.
- [ ] Any backwards-incompatible/breaking change has been clearly documented in the upgrading document.

### CODE REVIEW

File: autonomy/cli/helpers/deployment.py [hunk 1]
```diff
@@ -115,6 +142,31 @@ def run_deployment(
                 "SERVICE": None,
             }
         )
+        if detach:
+            click.echo("The service is running...")
+            return
+        click.echo("The service is running, press CTRL + C to stop...")
+        while True:
```
Comment (jmoreira-valory) [inline autonomy/cli/helpers/deployment.py:149]: Why is this here?

Comment (0xArdi) [reply autonomy/cli/helpers/deployment.py:149]: I think its to wait until the command deliberately stopped with CTRL + C

Comment (DavidMinarsch) [reply autonomy/cli/helpers/deployment.py:149]: So @angrybayblade we are now still blocking on the deploy command but no longer stream the logs? I think there can be a benefit for this but it can also be annoying. How about we control this behaviour with a flag? By default we exit cleanly and the user has to manually stop with the stop command. When the flag is set the command blogs and `CTRL + C` ensures exit.

Comment (DavidMinarsch) [reply autonomy/cli/helpers/deployment.py:149]: Maybe the default should be the current implementation. I let you decide.

**Review by 0xArdi — APPROVED** [review]
